lib/spack/docs/getting_started.rst

```diff
@@ -885,51 +894,78 @@ Environment Modules
 In order to use Spack's generated environment modules, you must have
 installed one of *Environment Modules* or *Lmod*.  On many Linux
 distributions, this can be installed from the vendor's repository.  For
-example: ``yum install environment-modules`` (Fedora/RHEL/CentOS).  If
-your Linux distribution does not have Environment Modules, you can get it
-with Spack:
+example: ``yum install environment-modules`` (Fedora/RHEL/CentOS). If
+your Linux distribution does not have Environment Modules, Spack can
+build it for you!
 
-#. Consider using system tcl (as long as your system has Tcl version 8.0 or later):
+What follows are three steps describing how to install and use environment-modules with spack.
 
-   #) Identify its location using ``which tclsh``
-   #) Identify its version using ``echo 'puts $tcl_version;exit 0' | tclsh``
-   #) Add to ``~/.spack/packages.yaml`` and modify as appropriate:
+#. Install ``environment-modules``.
 
-      .. code-block:: yaml
+   * Spack can build and install ``environment-modules`` for you.
+     Call ``spack bootstrap`` which will install ``environment-modules`` and it's 
+     dependencies.
 
-         packages:
-             tcl:
-                 paths:
-                     tcl@8.5: /usr
-                 buildable: False
+   * If you wish to install ``environment-modules`` in a more manual way,
+     try the following procedure:
 
-#. Install with:
+      * Consider using system tcl (as long as your system has Tcl version 8.0 or later):
```

Comment:
So, my comment about `tcl` is simply because `tcl` takes the longest to build and install in `environment-module`'s dependency stack. So since it is a common package to have already installed, I mention that you can get spack to use it instead of rebuilding it.

However.. Maybe this isn't a good idea to suggest because spack currently does not have a good method of interacting with native installed packages. Maybe I should just strike that bit about tcl entirely, thoughts?

Reply:
It looks like the comment about using system tcl was originally added by @citibeth. IMO it is worth keeping, but I think it should be added at the end of the section (especially if the only purpose is to speed up the installation): 

> You can speed up installation of `environment-modules` by referring Spack to use a system-installed tcl if it is available...

I'm not sure what you mean by:

> Maybe this isn't a good idea to suggest because spack currently does not have a good method of interacting with native installed packages.

Are you saying in your experience that specifying externals in packages.yaml has caused problems for you?

Reply:
No, the mechanics of adding externals seems to be okay for now.

What I worry about is a system `/usr/lib` contaminating the link line somewhere and linking against system packages on accident where spack packages should be used. I don't yet have a specific example when this has caused problems _yet_, but I feel that perhaps for stuff we're claiming will work well for all spack users we shouldn't loudly proclaim this functionality when it may cause some problems in some cases.

Reply:
There has been work to prevent system paths like `/usr/lib` from making their way into link paths. External packages are already linked in the guide on getting started. So I think it's worth keeping this. I do think it should be put at the end of the section though (not as part of the 3 steps), since it is easier (if longer) not to point Spack to the system tcl.